metis-preview/metis-preview-service/src/main/java/eu/europeana/metis/preview/service/ZipService.java

```diff
@@ -25,32 +24,31 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(ZipService.class);
 
-  public List<String> readFileToStringList(InputStream inputStream) throws ZipFileException {
+  public List<String> readFileToStringList(MultipartFile file) throws ZipFileException {
 
-    //TODO use a different zip library based on streams so we don't have to create files
-    //and delete them afterward
-    List<String> xmls = new ArrayList<>();
+    List<String> records = new ArrayList<>();
+    
     try {
-      String prefix = String.valueOf(new Date().getTime());
-      File tempFile = File.createTempFile(prefix, ".zip");
-      FileUtils.copyInputStreamToFile(inputStream, tempFile);
-      LOGGER.info("Temp file: {} created.", tempFile);
+    	String fileName = "/tmp/" + file.getName() + "/" + new Date().getTime();
+        FileUtils.copyInputStreamToFile(file.getInputStream(), new File(fileName + ".zip"));
 
-      ZipFile zipFile = new ZipFile(tempFile);
-      File unzippedDirectory = new File(tempFile.getParent(), prefix + "-unzipped");
-      zipFile.extractAll(unzippedDirectory.getAbsolutePath());
-      LOGGER.info("Unzipped contents into: {}", unzippedDirectory);
-      FileUtils.deleteQuietly(tempFile);
-      File[] files = unzippedDirectory.listFiles();
-
-      for (File input : files) {
-        xmls.add(IOUtils.toString(new FileInputStream(input), "UTF-8"));
-      }
-      FileUtils.deleteQuietly(unzippedDirectory);
+        ZipFile zipFile = new ZipFile(fileName + ".zip");
+        zipFile.extractAll(fileName);
+        FileUtils.deleteQuietly(new File(fileName + ".zip"));
+        File[] files = new File(fileName).listFiles();
+        
+        for (File input : files) {
+            if(!input.isDirectory()){
+                FileInputStream stream = new FileInputStream(input);
+                records.add(IOUtils.toString(stream));
+                stream.close();
+            }
+        }
```

Comment:
The unzippedFiles are not removed as it was performed previously.

Reply:
See line 40 in ZipService: FileUtils.deleteQuietly(new File(fileName + ".zip"));

Reply:
The line you are referring to deletes the .zip file and not the unzipped files in the new directory created.

Reply:
My bad. I've fixed this.